### ISSUE TITLE
[ML-DataFrame] GET Transform with a unknown ID does not return a 404

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#40003]
`GET /_data_frame/transforms/a_transform_that_doesn't_exist`

Does not return a 404 instead an empty response is returned. More dubiously the log is filled with many error messages 

```
[2019-03-13T16:23:01,545][ERROR][o.e.t.TransportService   ] [node-0] failed to handle exception for action [indices:data/read/get[s]], handler [org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler/org.elasticsearch.action.support.single.shard.TransportSingleShardAction$AsyncSingleAction$2@72a4f821]
java.lang.RuntimeException: Failed to load data frame transform configuration for transform [test-get]
```

```
[2019-03-13T16:23:01,545][WARN ][o.e.x.c.s.SchedulerEngine] [node-0] listener failed while handling triggered event [data_frame/transforms/schedule_test-get]
java.lang.NullPointerException: null
        at org.elasticsearch.xpack.dataframe.transforms.DataFrameTransformTask$ClientDataFrameIndexer.maybeTriggerAsyncJob(DataFrameTransformTask.java:272) ~[?:?]
        at org.elasticsearch.xpack.dataframe.transforms.DataFrameTransformTask.triggered(DataFrameTransformTask.java:189) ~[?:?]
```

```
Caused by: java.lang.RuntimeException: Failed to load data frame transform configuration for transform [test-get]
```

And finally after thousands of lines the cause
```
Caused by: org.elasticsearch.ResourceNotFoundException: Transform with id [test-get] could not be found
        at org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.lambda$getTransformConfiguration$3(DataFrameTransformsConfigManager.java:106) ~[?:?]
```

**Comment by hendrikmuhs** [comment]
This looks like 2 issues:

- non-404: sounds like a regression to me, which is strange as it's covered by a test, I need to check your circumstances 
- failure spam: known issue about failure handling

**Comment by davidkyle** [comment]
Closing this issue as it was fixed in #39934

**Comment by davidkyle** [comment]
@hendrikmuhs Ignore this sorry I was working on a old commit
